### PR TITLE
bin: Warn if config options are missing

### LINKED ISSUE #59: Relax configuration validation
To allow for better configuration options we need to temprarily relax our current configration validation until we have time to improve it.
One solution could be to warn the user that there might be something wrong in their confiuration and give them the option to abort and double check their configuration.

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:
One possible solution on how we can relax the configuration validation.

fixes #59 

**Special notes for reviewer**:
I am not sure if it is smart to read user input or not. Might not be so good for the pipeline.

This is what it looks like right now:
```
[ck8s] Validating wc config
[ck8s] WARN: fluentd.nodeSelector."node-role.kubernetes.io/master" is not set in /home/user/.ck8s/cloud_p/env_name/wc-config.yaml
[ck8s] Do you want to exit? (y/n): y
```

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

### CODE REVIEW

File: bin/common.bash [hunk 1]
```diff
@@ -208,7 +212,7 @@ config_load() {
       exit 1
     fi
 
-    validate_version "$1"
+    #validate_version "$1"
```
Comment (danielharr) [inline bin/common.bash:215]: Should this be commented out?

File: bin/common.bash [hunk 2]
```diff
@@ -88,7 +88,6 @@ validate_version() {
     ck8s_version=$(yq r "${file}" 'global.ck8sVersion')
     if [[ -z "$ck8s_version" ]]; then
         log_error "ERROR: global.ck8sVersion is not set in ${file}"
-        will_exit="true"
```
Comment (danielharr) [inline bin/common.bash:91]: Should the version variable not be required?

**Review by danielharr — COMMENTED** [review]
Not sure that I'm a fan of the input requirement :thinking: 
Looks good in general though, just a few questions :slightly_smiling_face:

File: bin/common.bash (hunk 1 above)
Comment (OlleLarsson) [reply bin/common.bash:215]: Nope

File: bin/common.bash (hunk 2 above)
Comment (OlleLarsson) [reply bin/common.bash:91]: Oh, right, it should be changed to `maybe_exit` instead.

Comment (danielharr) [reply bin/common.bash:91]: Yeah, we should probably try to declare more variables as `local` to reduce their scope :slightly_smiling_face:

File: bin/common.bash [hunk 3]
```diff
@@ -136,18 +136,23 @@ validate_config() {
...
-                will_exit="true"
+                log_warning "WARN: ${opt} is not set in ${file}"
+                maybe_exit="true"
             fi
         done
-        if [[ "${will_exit}" == "true" ]]; then
-            exit 1
+
+        if [[ "${maybe_exit}" == "true" ]]; then
+            echo -n -e "[\e[34mck8s\e[0m] Do you want to exit? (y/n): " 1>&2
```
Comment (danielharr) [inline bin/common.bash:151]: If we're going to keep the input requirement, maybe use the word "abort" or rephrase to "Continue anyway?" instead of "exit"?

**Review by viktor-f — APPROVED** [review]
LGTM

**Review by danielharr — APPROVED** [review]
LGTM

**Pull request merged**